storage/remote/otlptranslator/prometheus/normalize_label_test.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2024 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheus
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestNormalizeLabel(t *testing.T) {
+	tests := []struct {
+		label     string
+		allowUTF8 bool
+		expected  string
+	}{
+		{"", false, ""},
+		{"label_with_special_chars!", false, "label_with_special_chars_"},
+		{"123label", false, "key_123label"},
+		{"123label", true, "123label"}, // UTF-8 allows numbers at the beginning
+		{"_label", false, "key_label"},
+		{"_label", true, "_label"}, // UTF-8 allows single underscores at the beginning
+		{"__label", false, "__label"},
+		{"label_with_utf8_字符", true, "label_with_utf8_字符"},
```

Comment:
can you add a test case with a label that has dots in it? just so people see what happens :)

Reply:
While extending the test here I've noticed that Japanese characters aren't sanitized correctly when UTF8 isn't allowed :O 

![image](https://github.com/user-attachments/assets/2d8f1911-a75e-4ef4-a440-9b9660abb81b)


![image](https://github.com/user-attachments/assets/6242d016-91a1-437a-b8f1-66267a76e56e)

![image](https://github.com/user-attachments/assets/e91785f4-d1f5-42a4-bc1c-d025f601618d)

Reply:
model.EscapeName might be a better function to call

Reply:
I had to invert the order of adding the "key_" prefix when using model.EscapeName, but that's alright :)